File: src/main/java/com/kapilkoju/nepse/data/todaysprice/TodaysPrice.java
package com.kapilkoju.nepse.data.todaysprice;

import lombok.Builder;
import lombok.Value;

import java.math.BigDecimal;

@Value
@Builder
class TodaysPrice {
    private String companyName;
    private Integer noOfTransactions;
    private BigDecimal maxPrice;
    private BigDecimal minPrice;
    private BigDecimal closingPrice;
    private Integer tradedShares;
    private BigDecimal amount;
    private BigDecimal previousClosing;
    private BigDecimal difference;
}

File: src/main/kotlin/com/kapilkoju/nepse/data/todaysprice/model/TodaysPrice.kt
package com.kapilkoju.nepse.data.todaysprice.model

import java.io.Serializable
import java.math.BigDecimal
import java.time.LocalDate
import java.time.LocalTime
import javax.persistence.Entity
import javax.persistence.Id
import javax.persistence.IdClass
import javax.persistence.Table

data class TodaysPriceId(val date: LocalDate? = null, val companyName: String? = null) : Serializable

@Entity
@Table(name = "todays_price")
@IdClass(TodaysPriceId::class)
data class TodaysPrice(
        @Id
        val date: LocalDate,
        @Id
        val companyName: String,
        val noOfTransactions: Int,
        val maxPrice: BigDecimal,
        val minPrice: BigDecimal,
        val closingPrice: BigDecimal,
        val tradedShares: Int,
        val amount: BigDecimal,
        val previousClosing: BigDecimal,
        val difference: BigDecimal,
        val valueDate: LocalDate,
        val valueTime: LocalTime
)